RxSwift/Observables/TakeUntil.swift

```diff
@@ -225,3 +251,65 @@ final private class TakeUntilPredicate<Element>: Producer<Element> {
         return (sink: sink, subscription: subscription)
     }
 }
+
+// MARK: - TakeUntilCompletable
+final private class TakeUntilCompletableSink<Observer: ObserverType>
+    : Sink<Observer>
+    , LockOwnerType
+    , ObserverType
+    , SynchronizedOnType {
+    typealias Element = Observer.Element
+    typealias Parent = TakeUntilCompletable<Element>
+
+    fileprivate let _parent: Parent
+
+    let _lock = RecursiveLock()
+
+
+    init(parent: Parent, observer: Observer, cancel: Cancelable) {
+        self._parent = parent
+        super.init(observer: observer, cancel: cancel)
+    }
+
+    func on(_ event: Event<Element>) {
+        self.synchronizedOn(event)
+    }
+
+    func _synchronized_on(_ event: Event<Element>) {
+        switch event {
+        case .next:
+            self.forwardOn(event)
+        case .error:
```

Comment:
Since the code is the same for both cases, would be an option to use `case .error, .completed:`?

Reply:
Thanks, fixed